packages/next/client/route-announcer.tsx

```diff
@@ -1,7 +1,22 @@
 import React from 'react'
 import { useRouter } from './router'
 
-export function RouteAnnouncer() {
+const nextjsRouteAnnouncerStyles: React.CSSProperties = {
+  border: 0,
+  clip: 'rect(0 0 0 0)',
+  height: '1px',
+  margin: '-1px',
+  overflow: 'hidden',
+  padding: 0,
+  position: 'absolute',
+  width: '1px',
+
+  // https://medium.com/@jessebeach/beware-smushed-off-screen-accessible-text-5952a4c2cbfe
+  whiteSpace: 'nowrap',
+  wordWrap: 'normal',
+}
+
+export const RouteAnnouncer = React.memo(() => {
```

Comment:
RouteAnnouncer doesn't have to be memoized since there's no prop. Hoisting style looks good, we can keep that part

Reply:
Does react child component re-render when parents component state change even when props are not passed ? (Doubt)

Reply:
The parents are mostly context provider. From react docs about [memo](https://reactjs.org/docs/react-api.html#reactmemo): 
> React.memo has a useState, useReducer or useContext Hook in its implementation, it will still rerender when state or context change.

So I think it will still re-render

Reply:
Ok thanks, so better to remove `memo` ?

Reply:
Sounds good, let's remove memo and we can keep hoisting the styles

Reply:
> Sounds good, let's remove memo and we can keep hoisting the styles

BTW, Can you please 🥺 say , is this PR is correct https://github.com/vercel/next.js/pull/39296 :)

Reply:
> Sounds good, let's remove memo and we can keep hoisting the styles

Done 🙌🏽